crates/pallet-messenger/src/lib.rs

```diff
@@ -465,19 +539,36 @@ mod pallet {
 
             let provides_tag = (msg.dst_domain_id, msg.channel_id, msg.nonce);
             Inbox::<T>::insert((xdm.src_domain_id, xdm.channel_id, msg.nonce), msg);
+            Self::deposit_event(Event::InboxMessage {
```

Comment:
Hmm, is the purpose of depositing an event to let the relayer know that there is a new inbox message? I feel like the event is sort of misused if so, for the event is typically used to notify the external world that something did happen on the chain, but here it's not for the unsigned validation stage.

Reply:
No, this is not for the relayer but rather for external world to know when an inbox message is validated and added to the Inbox queue for processing. Relayer will only care about the event that comes out of processing those messages

Reply:
Then I feel it's confusing. You not only generate these events on processing these messages successfully when the unsigned extrinsics are included in some block but also generate them while still validating them in the tx pool, at that point, the inbox message is not yet actually added to the inbox queue on the chain.

Reply:
So InboxMessage is emitted when a new message received was validated and added to the Inbox queue. We add this event in pre_dispatch of the Unsigned as we add the message to the queue there after validating. Since the state changes are discarded when the transaction is still in the pool but is retained only when it actually run the extrisinc during the apply_block phase so this is feels like a right place to emit them. In what context is this confusing for you ?

Reply:
I mean the events deposited here are inside the `validate_unsigned`, they are not useful if the state changes are discarded anyway.

Reply:
the state is not discarded when the extrinsic is about to be dispatched. It is discarded when extrinsic is not included in the block but still in the txn_pool and we dont care about this stage

Reply:
Looks like what you meant is `pre_dispatch` but not `validate_unsigned`.

Reply:
Ahh I see, you are correct! will change that

Reply:
Nope, that's not the same as when the `validate_unsigned` is invoked in the tx pool, the events are still deposited which is not useful. Current behavior: `validate_unsigned` => event deposited, `pre_dispatch` => event deposited, but the one making more sense is: `validate_unsigned` => no event deposited, `pre_dispatch` => event deposited.

Reply:
> the events are still deposited which is not useful.
Yes but why would that matter as the state is discarded and events are not even published over subscription. 

In this context using validate_unsigned is intentional since most of the verification logic same but happy to separate them

Reply:
It won't cause substantial issues indeed, just feel unnecessary and nice to void that :(.